src/timeman.cpp

```diff
@@ -32,41 +32,41 @@ namespace {
 
   enum TimeType { OptimumTime, MaxTime };
 
-  const int MoveHorizon   = 50;   // Plan time management at most this many moves ahead
-  const double MaxRatio   = 7.09; // When in trouble, we can step over reserved time with this ratio
-  const double StealRatio = 0.35; // However we must not steal time from remaining moves over this ratio
-
-
-  // move_importance() is a skew-logistic function based on naive statistical
-  // analysis of "how many games are still undecided after n half-moves". Game
-  // is considered "undecided" as long as neither side has >275cp advantage.
-  // Data was extracted from the CCRL game database with some simple filtering criteria.
-
-  double move_importance(int ply) {
-
-    const double XScale = 7.64;
-    const double XShift = 58.4;
-    const double Skew   = 0.183;
-
-    return pow((1 + exp((ply - XShift) / XScale)), -Skew) + DBL_MIN; // Ensure non-zero
-  }
-
   template<TimeType T>
-  int remaining(int myTime, int movesToGo, int ply, int slowMover) {
-
-    const double TMaxRatio   = (T == OptimumTime ? 1 : MaxRatio);
-    const double TStealRatio = (T == OptimumTime ? 0 : StealRatio);
-
-    double moveImportance = (move_importance(ply) * slowMover) / 100;
-    double otherMovesImportance = 0;
-
-    for (int i = 1; i < movesToGo; ++i)
-        otherMovesImportance += move_importance(ply + 2 * i);
-
-    double ratio1 = (TMaxRatio * moveImportance) / (TMaxRatio * moveImportance + otherMovesImportance);
-    double ratio2 = (moveImportance + TStealRatio * otherMovesImportance) / (moveImportance + otherMovesImportance);
-
-    return int(myTime * std::min(ratio1, ratio2)); // Intel C++ asks for an explicit cast
+  int remaining(int myTime, int myInc, int moveOverhead, int movesToGo, int ply)
+  {
+    double TRatio, sd = 8.5;
+    int mn = (ply + 1) / 2; // current move number for any side
+
+    /// In movestogo case we distribute time according to quadratic function with the maximum around move 20 for 40 moves in y time case.
+ 
+    if (movesToGo)
+    {
+        TRatio = (T == OptimumTime ? 1.0 : 6.0) / movesToGo;
+        if (mn <= 40)
+            TRatio *= (1.1 - 0.001 * (mn-20) * (mn-20));
+        else
+            TRatio *= 1.5;
+    }
+    else
+    {
+        /// In sudden death case we increase usage of remaining time as the game goes on. This is controlled by parameter sd.
+
+        sd = 1.0 + 20.0 * mn / (500.0 + mn);
+        TRatio = (T == OptimumTime ? 0.017 : 0.07) * sd;
+    }
+    
+    /// In the case of no increment we simply have ratio = std::min(1.0, TRatio); The usage of increment follows quadratic distribution with the maximum at move 25.
+    
+    double incUsage = 0.0;
+
+    if (myInc) 
+        incUsage = std::max(55.0, 120.0 - 0.12 * (mn-25) * (mn-25));
+
+    double ratio = std::min(1.0, TRatio * (1.0 + incUsage * myInc / (myTime * sd)));
```

Comment:
So, the reason is that time management is always called in init, also for go depth as in bench. In that case myTime=0. 
```
#4  0x00000000004505d0 in (anonymous namespace)::remaining<(<unnamed>::TimeType)0u>(int, int, int, int, int) (myTime=0, myInc=0, moveOverhead=30, movesToGo=0, ply=0) at timeman.cpp:39
#5  0x00000000004504a5 in TimeManagement::init (this=0x7a2730 <Time>, limits=..., us=WHITE, ply=0) at timeman.cpp:109
#6  0x00000000004395f1 in MainThread::search (this=0x7ffff6b45010) at search.cpp:244
#7  0x000000000044a356 in Thread::idle_loop (this=0x7ffff6b45010) at thread.cpp:114
```

```